### PR TITLE
Update boundary system to use MRTK type instead of Unity type

### PR DESCRIPTION
**Pull request by keveleigh**
## Overview
In preparation for Unity's future boundary types, this PR looks to remove dependencies on Unity types.

### CODE REVIEW

File: Assets/MixedRealityToolkit/Definitions/BoundarySystem/BoundaryType.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.MixedReality.Toolkit.Boundary
+{
+    public enum BoundaryType
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit/Definitions/BoundarySystem/BoundaryType.cs]: Can you add doc comments to this enum and values?

Comment (keveleigh) [reply Assets/MixedRealityToolkit/Definitions/BoundarySystem/BoundaryType.cs]: Added!

**Pull request merged**